browser/js/dashboard/newGraph/newGraph.controller.js

```diff
@@ -17,7 +20,22 @@ app.controller('newGraphCtrl', function ($scope, $q, WidgetSettingsFactory, Gene
     	if(!!$scope.form.yparam) {
     		var ytype = $scope.form.yparam.type
     	}
-    	$scope.validGraphTypes = validGraphFactory.getValidGraphTypes (xtype, ytype)
+    	$scope.validGraphTypes = validGraphFactory
+            .getValidGraphTypes (xtype, ytype)
+            .map(function (type) {
+                let fakeWidget = {
+                    type: type,
+                    xparam: $scope.form.xparam.name,
+                    yparam: $scope.form.yparam.name
+                }
+
+                return {
+                    name: type,
+                    options: returnGraphOptions(type, $scope.form.xparam.name, $scope.form.yparam.name),
+                    data: DashboardFactory.setDataInCorrectFormat($scope.data, fakeWidget)
+                }
+            })
+
```

Comment:
Especially since it looks like the validGraphFactory.getValidGraphTypes method is only used in this file, might as well send it the $scope.form.xparam.name, $scope.form.yparam.name, and the $scope.data and have the factory return to the $scope the final mapped array so you don't have to do it here in the controller.